consensus/fork_choice/src/fork_choice.rs

```diff
@@ -1499,6 +1532,110 @@ where
     }
 }
 
+fn process_justification_and_finalization_from_progressive_cache<E, T>(
+    state: &BeaconState<E>,
+    spec: &ChainSpec,
+    progressive_balances_mode: ProgressiveBalancesMode,
+    log: &Logger,
+) -> Result<JustificationAndFinalizationState<E>, Error<T::Error>>
+where
+    E: EthSpec,
+    T: ForkChoiceStore<E>,
+{
+    let justification_and_finalization_state = JustificationAndFinalizationState::new(state);
+    if state.current_epoch() <= E::genesis_epoch() + 1 {
+        return Ok(justification_and_finalization_state);
+    }
+
+    // Load cached balances
+    let progressive_balances_cache: &ProgressiveBalancesCache = state.progressive_balances_cache();
+    let previous_target_balance =
+        progressive_balances_cache.previous_epoch_target_attesting_balance()?;
+    let current_target_balance =
+        progressive_balances_cache.current_epoch_target_attesting_balance()?;
+    let total_active_balance = state.get_total_active_balance()?;
+
+    if progressive_balances_mode.perform_comparative_checks() {
+        let participation_cache =
+            ParticipationCache::new(state, spec).map_err(Error::ParticipationCacheBuild)?;
+
+        if let Err(e) = check_progressive_balances::<E, T>(
+            state,
+            &participation_cache,
+            previous_target_balance,
+            current_target_balance,
+            total_active_balance,
+        ) {
+            return if progressive_balances_mode == ProgressiveBalancesMode::Strict {
+                // if comparative check fails in `Strict` mode, return error
+                Err(e)
+            } else {
+                error!(log,
+                    "Progressive balances mismatch, falling back to epoch processing calculation.";
+                    "error" => ?e,
+                );
```

Comment:
For the sake of log parsing in ELK we're now avoiding logs with commas

```suggestion
                error!(
                    log,
                    "Progressive balances mismatch";
                    "info" => "falling back to epoch processing calculation",
                    "error" => ?e,
                );
```